torch/distributed/distributed_c10d.py

```diff
@@ -2812,6 +2812,7 @@ def reduce(tensor, dst, op=ReduceOp.SUM, group=None, async_op=False):
         work.wait()
 
 
+@_time_logger
 def _object_to_tensor(obj, device, group):
     f = io.BytesIO()
     _pickler(f).dump(obj)
```

Comment:
If you care about pickling time here, serialize it with `protocol=pickle.HIGHEST_PROTOCOL`, as that will prevent an extra copy in a lot of cases.

Reply:
oh really, thanks for your suggestion.

Reply:
@fduwjj Added the arg as a suggestion

Reply:
@fduwjj Just checked and default PICKLE version is now 4. This should update it to 5 which is a mild, but potentially free speed up for objects which support it (like numpy arrays etc).

Reply:
Also pretty sure torch.multiprocessing already does this protocol.HIGHEST with our monkeypatch wrapper :)

Reply:
I see. Somehow I tried this protocol in our other applications, I didn't see any perf improvements.

Reply:
@fduwjj It will only really speed up things if the Python objects implement the new pickle protocol APIs to take advantage of it. It's really what should be done though in case where the pickle is for ephemeral process to process sharing like here.